src/generators/operationGenerator.ts

```diff
@@ -286,23 +291,47 @@ type ParameterWithDescription = OptionalKind<
 >;
 
 /**
- * Write operations implementation, extracted from OperationGroupDetails, to the generated file
+ * Gets a list of parameter declarations for each overload the operation supports,
+ * and the list of parameter declarations for the base operation.
  */
-export function writeOperations(
-  operationGroupDetails: OperationGroupDetails,
-  operationGroupClass: ClassDeclaration,
-  importedModels: Set<string>,
+function getOperationParameterSignatures(
+  operation: OperationDetails,
   parameters: ParameterDetails[],
-  isInline = false
+  importedModels: Set<string>,
+  operationGroupClass: ClassDeclaration
 ) {
-  operationGroupDetails.operations.forEach(operation => {
-    const responseName = getResponseType(operation, importedModels);
-    const paramDeclarations = filterOperationParameters(
-      parameters,
-      operation
-    ).map<ParameterWithDescription>(param => {
+  const operationParameters = filterOperationParameters(parameters, operation, {
+    includeContentType: true
+  });
+
+  const operationRequests = operation.requests;
+  const overloadParameterDeclarations: ParameterWithDescription[][] = [];
+  const hasMultipleOverloads = Boolean(operationRequests.length > 1);
+
+  for (const request of operationRequests) {
+    const requestMediaType = request.mediaType;
+    // filter our parameters that belong to a different media type
+    const requestParameters = operationParameters.filter(param => {
+      if (!param.targetMediaType) {
```

Comment:
If you want to make this filter shorter, you can have
```typescript
.filter(({targetMediaType}) => !targetMediaType || requestMediaType === targetMediaType)
```

Reply:
Shorter and more readable! Thanks, I made this change.